kornia/augmentation/_2d/mix/transplantation.py

```diff
@@ -0,0 +1,261 @@
+        tensor([[[0, 0, 1, 1, 0],
+                 [1, 2, 0, 0, 0],
+                 [1, 2, 1, 1, 0],
+                 [0, 0, 0, 0, 2],
+                 [2, 2, 2, 0, 2]],
+        <BLANKLINE>
+                [[2, 0, 0, 2, 1],
+                 [2, 1, 0, 2, 1],
+                 [2, 0, 1, 0, 2],
+                 [2, 2, 2, 0, 2],
+                 [2, 1, 0, 0, 0]]])
+        >>> image_out, mask_out = aug(image, mask)
+        >>> image_out.shape
+        torch.Size([2, 3, 5, 5])
+        >>> mask_out.shape
+        torch.Size([2, 5, 5])
+        >>> mask_out
+        tensor([[[2, 0, 1, 2, 0],
+                 [2, 2, 0, 2, 0],
+                 [2, 2, 1, 1, 2],
+                 [2, 2, 2, 0, 2],
+                 [2, 2, 2, 0, 2]],
+        <BLANKLINE>
+                [[0, 0, 0, 2, 0],
+                 [2, 1, 0, 0, 0],
+                 [2, 0, 1, 0, 0],
+                 [0, 0, 0, 0, 2],
+                 [2, 1, 0, 0, 0]]])
+
+    You can apply the same augmentation again, but the new mask should contain the same labels for the augmentation
+    to have an effect:
+
+        >>> aug._params["selected_labels"]  # Image 0 received label 2 from image 1 and image 1 label 0 from image 0
+        [tensor(2), tensor(0)]
+        >>> mask2 = torch.randint(0, 3, (2, 5, 5))
+        >>> image_out2, mask_out2 = aug(image, mask2, params=aug._params)
+        >>> mask_out2
+        tensor([[[2, 2, 2, 0, 1],
+                 [1, 2, 0, 1, 1],
+                 [0, 0, 1, 2, 2],
+                 [0, 0, 1, 2, 1],
+                 [1, 1, 1, 2, 0]],
+        <BLANKLINE>
+                [[0, 2, 2, 0, 0],
+                 [0, 0, 0, 0, 1],
+                 [0, 0, 0, 2, 0],
+                 [0, 0, 1, 0, 1],
+                 [1, 1, 1, 2, 0]]])
+    """
+
+    def __init__(
+        self,
+        excluded_labels: Optional[Sequence[int]] = None,
+        p: float = 0.5,
+        p_batch: float = 1.0,
+        data_keys: Optional[List[Union[str, int, DataKey]]] = None,
+    ) -> None:
+        super().__init__(p=p, p_batch=p_batch)
+
+        if excluded_labels is None:
+            excluded_labels = []
+        self.excluded_labels = tensor(excluded_labels)
+        if self.excluded_labels.ndim != 1:
+            raise ValueError(
+                f"excluded_labels must be a 1-dimensional sequence, but got {self.excluded_labels.ndim} dimensions."
+            )
+
+        if data_keys is None:
+            data_keys = [DataKey.INPUT, DataKey.MASK]
+        self.data_keys = [DataKey.get(inp) for inp in data_keys]
+        self._channel_dim = 1
+
+    def apply_non_transform_mask(self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any]) -> Tensor:
+        return input
+
+    def transform_input(self, acceptor: Tensor, donor: Tensor, selection: Tensor) -> Tensor:
+        # Expand selection to the channel dimension
+        selection = selection.unsqueeze(dim=self._channel_dim).expand_as(donor)
+        acceptor[selection] = donor[selection]
```

Comment:
This is an in-place operation. I think we shall try avoiding it?

Reply:
It is but the assignment happens on a cloned tensor. I could move the `clone()` to the `transform_*` functions, if you prefer.